### DISCUSSION TITLE
Boefjes blijven op status "Dispatched" staan

### DISCUSSION THREAD

**Discussion by AlexanderICS** [minvws/nl-kat-coordination#62] [Q&A]
Ik probeer het boefje "Nmap TCP (Top 250)" los te laten op een IPAddressV4 object, met daarin een geldig IP adres. De boefjes lijken nooit voorbij status "Dispatched" te komen. 

Het IP adres (192.168.2.2) is wel te scannen met Nmap als ik dit vanuit een terminal doe. Waarom komen de taken niet voorbij deze status? Het opnieuw installeren van OpenKAT heb ik al een keer geprobeerd (met een verse Debian install), alleen ik blijf ditzelfde probleem houden.

Flower zou de status kunnen tonen voor de Celery queues, alleen lijkt deze niet op de standaard poort 5555 te draaien. Kan ik hier wel bij?

**Comment by underdarknl** [comment]
zou je de output logs van bytes en de boefjes runner kunnen tonen?
Het is ons opgevallen dat soms de output-bestanden niet opgeslagen kunnen worden in bytes omdat de map /app/bytes/bytes-data niet bestaat of schrijfbaar is.

**Reply by AlexanderICS** [reply]
@underdarknl Waar staan de output logs van beiden? De logs van de Docker containers kon ik het niet in vinden, en in /var/log/ ook niet.

**Reply by AlexanderICS** [reply]
@underdarknl De logs van de bytes zouden in een bestand genaamd "bytes.log" moeten staan zie ik in de configuratie van de logs. Dit bestand is nergens beschikbaar op mijn harde schijf.

Is er ergens documentatie over de logging?

**Reply by underdarknl** [reply]
Je zou met docker logs nl-kat-coordination-boefjes-1 (of iets dat hier sterk op lijkt) eventuele runtime fouten van je boefjes moeten zijn. Een andere optie is mogelijk dat bytes tijdens het opslaan iets geks doet.  Ook hier zou je de logs op een zelfde manier moeten kunnen inzien.
Dispatched is de status waarin de worker de opdracht heeft opgepikt. De uitvoer gaat dan naar bytes waarna de scheduler zou moeten inzien dat het gelukt of gefaald is. Een hangende dispatched status betekend dus dat de runner (in de boefjes container) zijn werk niet af kan maken, maar ook niet crasht op een verwachte manier) of dat het werk niet bij bytes aankomt (omdat die potentieel struikelt).
